File: go/pkg/security/encryption.go
package security

import (
	"crypto/aes"
	"crypto/cipher"
	"fmt"
)

func encrypt(data []byte, key []byte, nonce []byte) ([]byte, error) {
	// encrypt
	block, err := aes.NewCipher(key)
	if err != nil {
		return nil, fmt.Errorf("could not create AES cipher: %s", err)
	}

	aesgcm, err := cipher.NewGCM(block)
	if err != nil {
		return nil, fmt.Errorf("could not create GCM: %s", err)
	}

	ciphertext := aesgcm.Seal(nil, nonce, data, nil)
	return ciphertext, nil
}

func decrypt(ciphertext, key, nonce []byte) ([]byte, error) {
	block, err := aes.NewCipher(key)
	if err != nil {
		return nil, fmt.Errorf("could not create AES cipher: %s", err)
	}

	aesgcm, err := cipher.NewGCM(block)
	if err != nil {
		return nil, fmt.Errorf("could not create GCM: %s", err)
	}

	plaintext, err := aesgcm.Open(nil, nonce, ciphertext, nil)
	if err != nil {
		return nil, fmt.Errorf("could not decrypt message: %s", err)
	}

	return plaintext, nil
}

File: python/util/encryption.py
from Cryptodome.Cipher import AES

def encrypt(data, rsa_key, aes_key):
    cipher = AES.new(aes_key, AES.MODE_GCM)
    nonce = cipher.nonce

    ciphertext, tag = cipher.encrypt_and_digest(data)
    ciphertext_and_tag = ciphertext + tag
    return ciphertext_and_tag, nonce

def decrypt(ctxt, aes_key, nonce):
    cipher = AES.new(aes_key, AES.MODE_GCM, nonce=nonce)
    ciphertext, tag = ctxt[:-16], ctxt[-16:]
    plaintext = cipher.decrypt_and_verify(ciphertext, tag)
    return plaintext

